MWSE/TES3Inventory.cpp

```diff
@@ -137,13 +137,13 @@ namespace TES3 {
 		return stack;
 	}
 
-	const auto TES3_Inventory_AddItem = reinterpret_cast<int(__thiscall*)(Inventory*, MobileActor *, Item *, int, bool, ItemData **)>(0x498530);
-	int Inventory::addItem(MobileActor * mobile, Item * item, int count, bool overwriteCount, ItemData ** itemDataRef) {
+	const auto TES3_Inventory_AddItem = reinterpret_cast<int(__thiscall*)(Inventory*, MobileActor *, PhysicalObject *, int, bool, ItemData **)>(0x498530);
+	int Inventory::addItem(MobileActor * mobile, PhysicalObject * item, int count, bool overwriteCount, ItemData ** itemDataRef) {
 		return TES3_Inventory_AddItem(this, mobile, item, count, overwriteCount, itemDataRef);
 	}
 
-	const auto TES3_Inventory_AddItemWithoutData = reinterpret_cast<int(__thiscall*)(Inventory*, MobileActor *, Item *, int, bool)>(0x497CD0);
-	int Inventory::addItemWithoutData(MobileActor * mobile, Item * item, int count, bool something) {
+	const auto TES3_Inventory_AddItemWithoutData = reinterpret_cast<int(__thiscall*)(Inventory*, MobileActor *, PhysicalObject *, int, bool)>(0x497CD0);
+	int Inventory::addItemWithoutData(MobileActor * mobile, PhysicalObject * item, int count, bool something) {
 		return TES3_Inventory_AddItemWithoutData(this, mobile, item, count, something);
```

Comment:
Unfortunately this function (and the one above) is not smart enough to understand leveled items. It won't resolve them correctly. Any leveled item resolving needs to be done sooner than this.

Reply:
The idea was to be able to add actual LEVI records to container inventories, without resolving them first like `tes3.addItem` does. I [used this in my ESR mod](https://github.com/MWSE/morrowind-nexus-lua-dump/blob/aad4ad1f9a18ece79c1bd33be67e685c6e3d2148/lua/Enchantment%20Services%20Redone%20-%20ESR/Enchantment%20Services%20Redone%20-%20ESR-51249-0-9-5-1670070494/mwse/mods/Virnetch/enchantmentServicesRedone/items/containers.lua#L95-L103) to add leveled item records to containers created through `tes3.createObject` and it worked fine prior to the commit that changed this to limit to TES3Items.

It might need some checks to prevent LEVIs from being added to NPC inventories though, I think I remember when testing this that adding them to the player's inventory broke some things. But pretty sure the same can happen when using the mwscript additem command, and I'm not sure if it's possible to check the owner of an inventory from here.

Reply:
Ah, that's fair. I think the best way to handle this is to check if the object has been cloned already. For example:
```cpp
if (mact || getActor()->isClone()) {
	// Actor has already had leveled items resolved. Resolve the leveled item and add the real item if one was rolled.
}
else {
	// Actor has not been resolved. It is safe to add a leveled item directly.
}
```